starlark/eval.go

```diff
@@ -119,6 +119,11 @@ type Frame struct {
 	locals   []Value         // local variables, for debugger
 }
 
+// NewFrame returns a new frame with the specified parent and callable.
```

Comment:
Maybe explain more about when this would be used? Sounds like there is a specific purpose from the PR description.

Reply:
Done: 
```
// It may be used to synthesize stack frames for error messages.
// Few clients should need to use this function.
```
Basically, these are not the droids you are looking for.